homeassistant/components/airvisual/config_flow.py

```diff
@@ -68,34 +69,23 @@ def async_get_options_flow(config_entry):
         if not user_input:
             return await self._show_form()
 
-        await self._async_set_unique_id(user_input[CONF_API_KEY])
+        geo_id = async_get_geography_id(user_input)
+        await self._async_set_unique_id(geo_id)
 
         websession = aiohttp_client.async_get_clientsession(self.hass)
         client = Client(websession, api_key=user_input[CONF_API_KEY])
 
-        try:
-            await client.api.nearest_city()
-        except InvalidKeyError:
-            return await self._show_form(errors={CONF_API_KEY: "invalid_api_key"})
+        if not self._api_key_checked:
+            # If the user configures multiple geographies via configuration.yaml, we
+            # only need to determine the API key's validity once:
+            try:
+                await client.api.nearest_city()
+            except InvalidKeyError:
+                return await self._show_form(errors={CONF_API_KEY: "invalid_api_key"})
 
-        data = {CONF_API_KEY: user_input[CONF_API_KEY]}
-        if user_input.get(CONF_GEOGRAPHIES):
-            data[CONF_GEOGRAPHIES] = user_input[CONF_GEOGRAPHIES]
-        else:
-            data[CONF_GEOGRAPHIES] = [
-                {
-                    CONF_LATITUDE: user_input.get(
-                        CONF_LATITUDE, self.hass.config.latitude
-                    ),
-                    CONF_LONGITUDE: user_input.get(
-                        CONF_LONGITUDE, self.hass.config.longitude
-                    ),
-                }
-            ]
+            self._api_key_checked = True
```

Comment:
This won't work. We instantiate a new config flow for each invocation. 

Instead, you should store it in hass.data. 

```python
checked_keys = self.hass.data.setdefault('airvisual_checked_api_keys', set())
```